google-cloud-spanner/lib/google/cloud/spanner/pool.rb

```diff
@@ -88,29 +154,43 @@ def reset
 
         def close
           @thread.kill if @thread
-          @pool.each(&:delete_session)
-          @pool = []
-          @queue = []
+          @all_sessions.each(&:delete_session)
+          @all_sessions = []
+          @session_queue = []
+          @transaction_queue = []
 
           true
         end
 
         private
 
         def init
-          @pool = []
-          @queue = []
+          @all_sessions = []
+          @session_queue = []
+          @transaction_queue = []
           ensure_keepalive_thread!
-          @min.times { new_session! }
+          # init session queue
+          @min.times { session_queue << new_session! }
```

Comment:
This and the the line below will make init block for a long time if min is high. Can this be done in a separate thread so as to not block init.